### PR TITLE
Skip stp/ldp only for unwind portion of prolog/epilog

### LINKED ISSUE #81278: Generalize Arm64 ldr/str to ldp/stp optimization
https://github.com/dotnet/runtime/pull/77540 introduced a JIT peephole optimization to convert consecutive ldr/str instructions to ldp/stp. It was limited to avoiding this optimization when either of the two ldr/str represented lclvar, which requires more work to properly handle the GC effects.

See https://github.com/dotnet/runtime/pull/77540#pullrequestreview-1268515376 for more discussion.

This issue tracks generalizing the optimization to handle the lclvar cases.

Also, the optimization was restricted to not work in prologs or epilogs, to avoid affecting unwind codes. There are two potential improvements here:
1. Make sure the unwind codes work with the optimization.
2. Allow the optimization in the non-OS (not unwindable) part of the prolog/epilog.

### PR DESCRIPTION
**Pull request by kunalspathak**
This is the last portion remaining in https://github.com/dotnet/runtime/issues/81278. We will do peephole optimizations of replacing `ldr/str` with pair instructions `ldp/stp` in prolog as well. We will skip the portion of the prolog where we are generating code for unwind.

Fixes: https://github.com/dotnet/runtime/issues/81278

### CODE REVIEW

**Comment by kunalspathak** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=259463&view=ms.vss-build-web.run-extensions-tab) are great.

image

cc: @a74nh , @SwapnilGaikwad

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Review by BruceForstall — APPROVED** [review]
One suggestion

**Comment by kunalspathak** [comment]
gcstress is clean - https://dev.azure.com/dnceng-public/public/_build/results?buildId=259668&view=ms.vss-test-web.build-test-results-tab

File: src/coreclr/jit/unwindamd64.cpp [hunk 1]
```diff
@@ -86,6 +86,7 @@ short Compiler::mapRegNumToDwarfReg(regNumber reg)
 //
 void Compiler::unwindBegProlog()
 {
+    compGeneratingUnwindProlog = true;
```
[the remark this answers is not in the document]
Comment (kunalspathak) [reply src/coreclr/jit/unwindamd64.cpp:90]: done

**Comment by kunalspathak** [comment]
@BruceForstall - I didn't realize that I didn't push a change that renames `unwindarm.cpp` to `unwindarmarch.cpp`. Pushed it now.

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**